osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -123,21 +123,42 @@ public WorkingBeatmap CreateNew(RulesetInfo ruleset, User user)
                 Metadata = metadata,
                 Beatmaps = new List<BeatmapInfo>
                 {
-                    new BeatmapInfo
-                    {
-                        BaseDifficulty = new BeatmapDifficulty(),
-                        Ruleset = ruleset,
-                        Metadata = metadata,
-                        WidescreenStoryboard = true,
-                        SamplesMatchPlaybackRate = true,
-                    }
+                    BeatmapInfo.CreateDefault(ruleset, metadata)
                 }
             };
 
             var working = Import(set).Result;
             return GetWorkingBeatmap(working.Beatmaps.First());
         }
 
+        /// <summary>
+        /// Adds a beatmap to an existing set.
+        /// </summary>
+        /// <param name="beatmapSet">The beatmap set to add the <paramref name="beatmap"/> to.</param>
+        /// <param name="beatmap">The beatmap to be added.</param>
+        /// <param name="beatmapSkin">The skin specific to the beatmap being added.</param>
+        /// <returns>The usable <see cref="WorkingBeatmap"/> corresponding to <paramref name="beatmap"/> and associated with the supplied <paramref name="beatmapSet"/>.</returns>
+        /// <exception cref="InvalidOperationException">If <paramref name="beatmapSet"/> does not exist in the database.</exception>
+        public virtual WorkingBeatmap AddToExistingSet(BeatmapSetInfo beatmapSet, IBeatmap beatmap, ISkin beatmapSkin = null)
```

Comment:
I am suddenly regretting not merging this PR before my recent changes..

Reply:
I'll close and rebase on top whatever the new structure ends up being. No worry.

Reply:
Either way works. Was more of a joke than a serious issue.